src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -261,22 +269,52 @@ public void cleanThing(IotClient client, ThingInfo thing, boolean deletePolicies
     }
 
     /*
-     * Download root CA to a local file if the file does not exist
+     * Download root CA to a local file.
+     *
+     * To support HTTPS proxies and other custom truststore configurations, append to the file if it exists.
      */
-    private void downloadRootCAToFile(File f) throws IOException {
+    private void downloadRootCAToFile(File f) {
         if (f.exists()) {
-            outStream.printf("Root CA found at \"%s\". Skipping download.%n", f);
-            return;
+            outStream.printf("Root CA file found at \"%s\". Contents will be preserved.%n", f);
         }
         outStream.printf("Downloading Root CA from \"%s\"%n", ROOT_CA_URL);
-        downloadFileFromURL(ROOT_CA_URL, f);
+        try {
+            downloadFileFromURL(ROOT_CA_URL, f, f.exists());
+            removeDuplicateCertificates(f);
+        } catch (IOException e) {
+            // Do not block as the root CA file may have been manually provisioned
+            outStream.printf("Failed to download Root CA - %s%n", e);
+        }
+    }
+
+    private void removeDuplicateCertificates(File f) {
+        try {
+            String certificates = IoUtils.toUtf8String(Files.newInputStream(f.toPath()));
```

Comment:
the input stream must be closed. Use try-with-resources

Reply:
Done